### DISCUSSION TITLE
Training virial from forces and energys without raw virial datas

### DISCUSSION THREAD

**Discussion by ald3hyde** [deepmodeling/deepmd-kit#2763] [Q&A]
Dear all,
I will try training NpT simulation to reproduce density, young modules, and etc. However, I don't have virial data, so I want to know how to calculate virials from forces, energies, and lattice parameters to fit the model with virial. Also, is the thermal effect included in the virial which DeepMD uses? I do not want to regenerate DFT data and would like to know your suggestions.

**Comment by njzjz** [comment]
The virial is defined in _R. Clausius, Phil. Mag. 40, 122-127 (1870)_. You can also refer to the references of DeePMD-kit _Comput. Phys. Comm. 228, 178-184 (2018)_ and  _J. Chem. Phys. 159, 054801 (2023)_, which contain the definition of virial.
